extensions/wikia/ArticleComments/classes/ArticleCommentList.class.php

```diff
@@ -916,63 +924,98 @@ static function formatList($comments) {
 		return $template->render( 'comment-list' );
 	}
 
-	/*
-	 * @access public
-	 * @deprecated don't use it for Oasis, still needed for non-Oasis skins
-	 * @deprecated - not used in Oasis
+	/**
+	 * @deprecated Don't use it for Oasis, still needed for non-Oasis skins
 	 *
 	 * @return String HTML text with rendered comments section
 	 */
-
 	public function render() {
-		return F::app()->renderView('ArticleComments', 'Index');
+		return F::app()->renderView( 'ArticleComments', 'Index' );
  	}
 
 	/**
-	 * static entry point for hook
+	 * Static entry point for hook
 	 *
 	 * @param Title $title
 	 * @param Article $article
 	 *
-	 * @static
-	 * @access public
+	 * @return bool
 	 */
-	static public function ArticleFromTitle( &$title, &$article ) {
-		if( MWNamespace::isTalk($title->getNamespace()) && ArticleComment::isTitleComment($title) ) {
-			global $wgRequest, $wgOut;
-			$redirect = $wgRequest->getText('redirect', false);
-			$diff = $wgRequest->getText('diff', '');
-			$oldid = $wgRequest->getText('oldid', '');
-			$action = $wgRequest->getText('action', '');
-			$permalink = $wgRequest->getInt( 'permalink', 0 );
-
-			if (($redirect != 'no') && empty($diff) && empty($oldid) && ($action != 'history') && ($action != 'delete')) {
-				$parts = ArticleComment::explode($title->getText());
-				$redirectTitle = Title::newFromText($parts['title'], MWNamespace::getSubject($title->getNamespace()));
-				$commentId = $title->getArticleID();
-
-				if ($redirectTitle) {
-					$query = array();
-
-					if ( $permalink || $commentId !== 0 ) {
-						if( $commentId !== 0 ) {
-							/** bugId:11179 @author: nAndy */
-							$permalink = $commentId;
-						}
+	static public function ArticleFromTitle( Title &$title, &$article ) {
+		// Don't bother checking for redirects if we're not loading this article for the current request
+		if ( !self::isTitleForCurrentRequest( $title ) ) {
```

Comment:
This is the actual fix for this ticket.  Some code earlier the request creates one or more article objects via ```Article::newFromTitle``` and this code was intercepting it.  If the title in question was a comment it was issuing a redirect to the wall or forum page, with a jump to the comment there.  This makes sense if the user actually wants to load the comment, but if the comment is being loaded to get its info, we shouldn't be redirecting the user.